[14.0][FIX+IMP] storage_image, storage_image_product and storage_image_product_brand

ivantodorovich opened the pull request:
Closes https://github.com/OCA/storage/issues/95

* Improve image_handle widgets drag&drop feature.
* Refactor code + remove dead code.
* New and improved base views for image.relation.abstract
* Improve tests

ivantodorovich committed:
[IMP] storage_image_product: Technical improvements

ivantodorovich committed:
[IMP] storage_image_product_brand: Technical improvements

ivantodorovich commented:
Actually no, still working on it :man_facepalming:

ivantodorovich commented:
Regarding: https://github.com/OCA/storage/issues/95

This thing played tricks on me.. at first I couldn't understand why it worked perfectly fine with `debug=assets`, but not without it.. then I realized odoo added their own `image_url` widget and it was conflicting with ours, depending on the script loading order :man_facepalming: 

So I ended up removing the widget alltogether. It doesn't allow to edit pictures when using it, but that's ok.. storage images are not meant to be edited. For create, we can simply use the image widget on the `data` field. So, on the `storage.image` form view, we're not using the `image_url` widget anymore.. anyway, the user doesn't go there often so it's a good tradeoff I think. Also this way we can remove the ugly hack on `storage.image::create`

I also renamed `image_handle` to `storage_image_handle` to avoid possible issues like this one

ivantodorovich committed:
[IMP] storage_image_import: Tests

Test didn't make any sense to me, and they were failing because
of changes made in storage_image.

I rewrote them in a way that makes sense, and removed the dependency
to storage_image_product which was only here for tests I believe.

ivantodorovich commented:
Also fixed `storage_image_import` tests that were failing after my refactor and didn't make much sense to me.
Rewrote them in a simpler way IMO, here: https://github.com/OCA/storage/pull/103/commits/717617b646525ee53aebb98d479558cf2a9a4604 also removing the dependency to `storage_image_product` that was only there for the previous tests cases

simahawk commented:
Please fwd port this before doing these changes https://github.com/OCA/storage/pull/94

simahawk commented on storage_image_import/tests/models/fake_image_relation.py:
```diff
@@ -0,0 +1,10 @@
+from odoo import models
+
+
+class FakeImageRelation(models.Model):
```
see how I used odoo-test-helper https://github.com/OCA/storage/pull/94 (to be fwd ported)

simahawk reviewed (changes requested):
pls split the image widget fix

ivantodorovich committed:
[FIX] storage_image: image_url widget (closes #95)

ivantodorovich committed:
[FIX] Remove missing .less field (dead code)

ivantodorovich committed:
[FIX] storage_image_product_brand: Do not store url

The URL for all storage files is normally computed via backend.
This ensures that if you have different conf by env
you won't mess up w/ images coming from other envs (eg: prod).

This change makes sure that the URL which is used for the thumbs
is always computed by the backend.

simahawk commented on storage_image/security/security.xml:
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
```
what's the purpose of merging 2 records in a single file? It's preferable to have the in 2 files as before.

ivantodorovich replied:
Oh, didn't think much about it. It's just standardization, as this is how you'd find it in the majority of modules.
I'll revert if you mind it.

ivantodorovich replied:
done!

simahawk reviewed (commented):
LG, can you restore the security files?

ivantodorovich committed:
[IMP] storage_image: Technical and usability improvements

* Improved kanban widget (scss + js)
* Align code to standards.
* Remove dead code.
* New base views for abstract.relation

sebastienbeau commented:
Hi @ivantodorovich

Thanks for the refactor, opening the pop-up is really better.

The only missing feature with the previous version is the feedback during the drag and drop
![image](https://user-images.githubusercontent.com/1164578/130828909-2711ec0c-4af2-4ee1-9cb4-6946c205c9fa.png)

Indeed when you drop the first image the zone where the drag and drop work is not so big and without feedback you can drop in in the wrong place and this will open the image in the browser

sebastienbeau commented:
@ivantodorovich the weird poijnt is that you have defined a css for the drop-zone here https://github.com/OCA/storage/pull/103/files#diff-0388f7ef60e004bc536459637a780f8aa1261f05ce7f0d8f0a8f383308e46cf0R9

ivantodorovich commented:
> @ivantodorovich the weird poijnt is that you have defined a css for the drop-zone here https://github.com/OCA/storage/pull/103/files#diff-0388f7ef60e004bc536459637a780f8aa1261f05ce7f0d8f0a8f383308e46cf0R9

Indeed, it seems I broke it by mistake.
I've just pushed the fix :) Thanks for catching it!

sebastienbeau reviewed (approved):
LGTM code review and test.
Thanks for the improvement and the quick fix

sebastienbeau commented:
/ocabot merge minor

sebastienbeau commented:
Now we just need to export the brand image in shopinvader module and we are done ;)
